### PR TITLE
Allow the parent window to configure and manage navigation requests

### PR DESCRIPTION
**Pull request by sherbastian**
Allow the parent window to configure and manage navigation requests when the NR editor resides in an embedded context/iframe

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

@closes https://github.com/FlowFuse/nr-launcher/issues/223

### CODE REVIEW

File: package.json
Comment (Steve-Mcl) [inline package.json:3]:
```suggestion
    "version": "2.2.2",
```

We dont bump version. This is dont when the release is cut.

File: lib/theme/common/forge-common.js
Comment (Steve-Mcl) [reply lib/theme/common/forge-common.js:9]: This is fine. The whole of this code is (deliberately) self contained within an IIFE and will remain in memory for the lifetime of the visit.

File: lib/theme/common/forge-common.js
Comment (Steve-Mcl) [reply lib/theme/common/forge-common.js:23]: An alternative would be to replace the anchor with a button or a span, but this works too.
Happy to leave as is.

File: package.json
Comment (Steve-Mcl) [reply package.json:3]: i looks like you have again bumped the package @sherbastian 

Please remove this change.

**Comment by sherbastian** [comment]
Reverted the v bump, the PR should be good to go. It can be released independently of [#3656](https://github.com/FlowFuse/flowfuse/issues/3656) without any side effects

**Comment by sherbastian** [comment]
@Steve-Mcl, should I do it manually?

**Comment by Steve-Mcl** [comment]
> @Steve-Mcl, should I do it manually?

Yes, just update the value in this PR. The inner package is not published and has no automations.

**Review by Steve-Mcl — APPROVED** [review]
Pulled and tested locally against current FF core. Working as expected. Happy to approve.

I won't merge as I don't know the status of the other parts of this puzzle (so just in case any tweaks are needed across the PRs). I'll leave it to you to merge

You can merge at will.

**Pull request merged**
